test/test_oneview_appliance_device_snmp_v1_trap_destinations.py

```diff
@@ -0,0 +1,180 @@
+    state='present',
+    data=dict(communityString=DEFAULT_SNMPv1_TRAP_TEMPLATE['communityString'])
+)
+
+PARAMS_WITH_CHANGES = dict(
+    config='config.json',
+    state='present',
+    data=dict(destination=DEFAULT_SNMPv1_TRAP_TEMPLATE['destination'],
+              communityString='private')
+)
+
+PARAMS_FOR_ABSENT = dict(
+    config='config.json',
+    state='absent',
+    data=dict(destination=DEFAULT_SNMPv1_TRAP_TEMPLATE['destination'])
+)
+
+PARAMS_FOR_ABSENT_WITH_URI = dict(
+    config='config.json',
+    state='absent',
+    data=dict(uri=DEFAULT_SNMPv1_TRAP_TEMPLATE['uri'])
+)
+
+
+@pytest.mark.resource(TestApplianceDeviceSnmpV1TrapDestinationsModule='appliance_device_snmp_v1_trap_destinations')
+class TestApplianceDeviceSnmpV1TrapDestinationsModule(OneViewBaseTest):
+
+    def test_should_create_new_snmp_v1_trap_destination(self):
+        self.resource.get_all.return_value = []
+        self.resource.create.return_value = DEFAULT_SNMPv1_TRAP_TEMPLATE
+
+        self.mock_ansible_module.params = PARAMS_FOR_PRESENT
+
+        ApplianceDeviceSnmpV1TrapDestinationsModule().run()
+
+        self.mock_ansible_module.exit_json.assert_called_once_with(
+            changed=True,
+            msg=ApplianceDeviceSnmpV1TrapDestinationsModule.MSG_CREATED,
+            ansible_facts=dict(appliance_device_snmp_v1_trap_destinations=DEFAULT_SNMPv1_TRAP_TEMPLATE)
+        )
+
+    def test_should_not_update_when_data_is_equals(self):
+        self.resource.get_by.return_value = [DEFAULT_SNMPv1_TRAP_TEMPLATE]
+
+        self.mock_ansible_module.params = PARAMS_FOR_PRESENT
+
+        ApplianceDeviceSnmpV1TrapDestinationsModule().run()
+
+        self.mock_ansible_module.exit_json.assert_called_once_with(
+            changed=False,
+            msg=ApplianceDeviceSnmpV1TrapDestinationsModule.MSG_ALREADY_PRESENT,
+            ansible_facts=dict(appliance_device_snmp_v1_trap_destinations=DEFAULT_SNMPv1_TRAP_TEMPLATE)
+        )
+
+    def test_should_get_the_same_resource_by_uri(self):
+        self.resource.get_by.return_value = [DEFAULT_SNMPv1_TRAP_TEMPLATE]
+
+        self.mock_ansible_module.params = PARAMS_FOR_PRESENT_WITH_URI
+
+        ApplianceDeviceSnmpV1TrapDestinationsModule().run()
+
+        self.mock_ansible_module.exit_json.assert_called_once_with(
+            changed=False,
+            msg=ApplianceDeviceSnmpV1TrapDestinationsModule.MSG_ALREADY_PRESENT,
+            ansible_facts=dict(appliance_device_snmp_v1_trap_destinations=(DEFAULT_SNMPv1_TRAP_TEMPLATE))
+        )
+
+    def test_should_fail_with_missing_required_attributes(self):
+        self.mock_ansible_module.params = PARAMS_FOR_INVALID
+
+        ApplianceDeviceSnmpV1TrapDestinationsModule().run()
+
+        self.mock_ansible_module.fail_json.assert_called_once_with(
+            exception=mock.ANY,
+            msg=ApplianceDeviceSnmpV1TrapDestinationsModule.MSG_VALUE_ERROR
+        )
+
+    def test_update_when_data_has_modified_attributes(self):
```

Comment:
Since this endpoint has been added starting from API600, we should restrict this test case for api version 600:
self.mock_ov_client.api_version = 600

Reply:
Added in commit 1021d46799565d783520d76cba59652d9d3575e9